src/heartbeat/heartbeat.c

```diff
@@ -196,45 +204,56 @@ void hb_tx_cb(uint8_t* data, uint8_t* len) {
+        }
 
         if (hb_send_obc_ping) {
-            data[0] = 1;
-            data[1] = hb_self_id;
-            data[2] = HB_OBC;
-        } else if (hb_send_eps_ping) {
-            data[0] = 1;
-            data[1] = hb_self_id;
-            data[2] = HB_EPS;
+            data[HB_SENDER] = hb_self_id;
+            data[HB_RECEIVER] = HB_OBC_PLACEHOLDER;
+            data[HB_OPCODE] = HB_PING_REQUEST;
         } else if (hb_send_pay_ping) {
-            data[0] = 1;
-            data[1] = hb_self_id;
-            data[2] = HB_PAY;
+            data[HB_SENDER] = hb_self_id;
+            data[HB_RECEIVER] = HB_PAY_PLACEHOLDER;
+            data[HB_OPCODE] = HB_PING_REQUEST;
+        } else if (hb_send_eps_ping) {
+            data[HB_SENDER] = hb_self_id;
+            data[HB_RECEIVER] = HB_EPS_PLACEHOLDER;
+            data[HB_OPCODE] = HB_PING_REQUEST;
+
         } else if (hb_send_obc_resp) {
-            data[0] = 2;
-            data[1] = hb_self_id;
-            data[2] = HB_OBC;
-        } else if (hb_send_eps_resp) {
-            data[0] = 2;
-            data[1] = hb_self_id;
-            data[2] = HB_EPS;
+            data[HB_SENDER] = hb_self_id;
+            data[HB_RECEIVER] = HB_OBC_PLACEHOLDER;
+            data[HB_OPCODE] = HB_PING_RESPONSE;
+            data[HB_RESTART_REASON] = restart_reason;
+            data[HB_RESTART_COUNT] = (restart_count & 0xFF000000) >> 24;
+            data[HB_RESTART_COUNT+1] = (restart_count & 0x00FF0000) >> 16;
+            data[HB_RESTART_COUNT+2] = (restart_count & 0x0000FF00) >> 8;
+            data[HB_RESTART_COUNT+3] = (restart_count & 0x000000FF);
         } else if (hb_send_pay_resp) {
-            data[0] = 2;
-            data[1] = hb_self_id;
-            data[2] = HB_PAY;
+            data[HB_SENDER] = hb_self_id;
+            data[HB_RECEIVER] = HB_PAY_PLACEHOLDER;
+            data[HB_OPCODE] = HB_PING_RESPONSE;
+            data[HB_RESTART_REASON] = restart_reason;
+            data[HB_RESTART_COUNT] = (restart_count & 0xFF000000) >> 24;
+            data[HB_RESTART_COUNT+1] = (restart_count & 0x00FF0000) >> 16;
+            data[HB_RESTART_COUNT+2] = (restart_count & 0x0000FF00) >> 8;
+            data[HB_RESTART_COUNT+3] = (restart_count & 0x000000FF);
+        } else if (hb_send_eps_resp) {
+            data[HB_SENDER] = hb_self_id;
+            data[HB_RECEIVER] = HB_EPS_PLACEHOLDER;
+            data[HB_OPCODE] = HB_PING_RESPONSE;
+            data[HB_RESTART_REASON] = restart_reason;
+            data[HB_RESTART_COUNT] = (restart_count & 0xFF000000) >> 24;
+            data[HB_RESTART_COUNT+1] = (restart_count & 0x00FF0000) >> 16;
+            data[HB_RESTART_COUNT+2] = (restart_count & 0x0000FF00) >> 8;
+            data[HB_RESTART_COUNT+3] = (restart_count & 0x000000FF);
         } else {
             // Should not get here
-            data[0] = 0;
-            data[1] = 0;
-            data[2] = 0;
+            print("Error: Failed to execute hb tx callback\n");
```

Comment:
For code size and memory reasons, don't add an extra print statement (I like the idea and normally recommend it, but in this case you can detect this error if you see the CAN message is all zeros).